File: v2/src/shared/connection.ts
export interface SshTunnel {
  host: string;
  port: number;
  user: string;
  keyPath: string;
}

export interface ConnectionInput {
  name: string;
  uri: string;
  ssh?: SshTunnel;
  notes?: string;
}

export interface ConnectionRecord {
  id: string;
  name: string;
  uri: string;
  ssh?: SshTunnel;
  notes?: string;
  createdAt: number;
  updatedAt: number;
  lastUsedAt?: number;
}

export interface ConnectionSummary {
  id: string;
  name: string;
  hasSsh: boolean;
  notes?: string;
  createdAt: number;
  updatedAt: number;
  lastUsedAt?: number;
}

File: v3/src/main/kotlin/io/mex/data/Connection.kt
package io.mex.data

data class ConnectionRecord(
    val id: String,
    val name: String,
    val uri: String,
    val notes: String?,
    val createdAt: Long,
    val updatedAt: Long,
    val lastUsedAt: Long?,
    /** DBA-RO-1 — every mutating affordance in the UI is disabled for this connection. */
    val readOnly: Boolean = false,
)

data class ConnectionSummary(
    val id: String,
    val name: String,
    val notes: String?,
    val createdAt: Long,
    val updatedAt: Long,
    val lastUsedAt: Long?,
    val readOnly: Boolean = false,
)

data class ConnectionInput(
    val name: String,
    val uri: String,
    val notes: String? = null,
    val readOnly: Boolean = false,
)

data class UriHistoryEntry(
    val id: Long,
    val uri: String,
    val preview: String,
    val usedAt: Long,
)
